opencog/query/VirtualMatch.cc

```diff
@@ -256,20 +266,12 @@ void PatternMatch::validate_clauses(std::set<Handle>& vars,
 		              __FUNCTION__);
 	}
 
-	bogus = remove_constants(vars, negations);
```

Comment:
this is a CPU waster, I know, but it appears that people sometimes misunderstand bindlink and write them with an invalid syntax.  What we really need is a syntax validator that runs only once, and checks the syntax. After that, the bindlink would not ever need to be checked again, before use ... 

The problem is that invalid syntax tends to cause these hard-to-spot, hard-to-debug crazy crashes ... one looses a lot of time crawling through C++ code, only to realize later that the bindlink syntax was bad ...